### PR TITLE
gcc compilation feature

### PR DESCRIPTION
**Pull request by sergstetsuk**
Not sure if it helps. But I managed to compile jwlink with gcc. There are some warnings and inconsistencies in sources. You can review code and merge into main branch.

### CODE REVIEW

**Review by milabs — CHANGES_REQUESTED** [review]
Quite complex and unclean patch. I would suggest to split whole the patch into the smaller parts.

File: c/mem.c [hunk 1]
```diff
@@ -35,6 +35,7 @@
 #ifdef __WATCOMC__
```
Comment (XVilka) [inline c/mem.c]: Just remove __WATCOMC__ then.

File: sdk/rc/wres/h/filefmt.h [hunk 2]
```diff
@@ -66,23 +66,23 @@ typedef struct WResExtHeader {  /* Only present if WResVer >= 1 */
 typedef struct WResIDName {             /* this is a "Pascal style" string */
     uint_16     NumChars;               /* length */
     char        Name[1];                /* followed by that many characters */
-} _WCUNALIGNED WResIDName;              /* NOTE: there is no trailing '\0' */
+} /*_WCUNALIGNED*/ WResIDName;              /* NOTE: there is no trailing '\0' */
```
Comment (XVilka) [inline sdk/rc/wres/h/filefmt.h]: why not to redefine `_WCUNALIGNED` instead?

File: c/mem.c [hunk 3]
```diff
@@ -189,7 +190,9 @@ void *LnkExpand( void *src, size_t size )
 #ifdef TRMEM
     return( _trmem_expand( src, size, _trmem_guess_who(), TrHdl ) );
 #else
-    return( _expand( src, size ) );
+    //todo: return( _expand( src, size ) );
```
Comment (XVilka) [inline c/mem.c]: just make it conditional maybe?

**Comment by XVilka** [comment]
Please address the comments I added.

**Comment by sergstetsuk** [comment]
Nice suggestions. Good job. I'll try

**Review by sergstetsuk — COMMENTED** [review]
Reviewed changes. Revert _WCUNALIGNED and O_BINARY.

File: c/libr.c [hunk 4]
```diff
@@ -86,6 +79,13 @@ static int ARCompIName( const void *key, const void *vbase );
 int (*CmpARRtn)( const void *, const void * ) = ARCompName;
 int (*CmpOMFRtn)( const void *, const void *, unsigned ) = memcmp;
 
+#if defined( _M_IX86 ) && defined(__WATCOMC__)
```
Comment (milabs) [inline c/libr.c:82]: What's the reason to move those lines from one place to other?

Comment (sergstetsuk) [reply c/libr.c:82]: It's just taking function definitions out of #ifdef:
ARCompName
ARCompIName
CmpARRtn
CmpOMFRtn
if _WATCOMC is not defined you'll get error without these functions

**Comment by sergstetsuk** [comment]
.gitignore fixed.
Actually GccUnixR/ needs to be ingnored

**Comment by XVilka** [comment]
Sorry for the delay, thank you a lot for the PR!

**Pull request merged**